website/docs/d/healthcare_medtech_service.html.markdown

```diff
@@ -0,0 +1,60 @@
+---
+subcategory: "Healthcare"
+layout: "azurerm"
+page_title: "Azure Resource Manager: azurerm_healthcare_medtech_service"
+description: |-
+  Get information about an existing Healthcare Med Tech Service
+---
+
+# Data Source: azurerm_healthcare_medtech_service
+
+Use this data source to access information about an existing Healthcare Med Tech Service
+
+## Example Usage
+
+```hcl
+data "azurerm_healthcare_medtech_service" "example" {
+  name         = "tfexmedtech"
+  workspace_id = "tfexwks"
+}
+
+output "azurerm_healthcare_medtech_service_id" {
+  value = data.azurerm_healthcare_medtech_service.example.id
+}
+```
+## Argument Reference
+
+* `name` - The name of the Healthcare Med Tech Service.
+
+* `workspace_id` - The id of the Healthcare Workspace in which the Healthcare Med Tech Service exists.
+
+## Attributes Reference
+
+The following attributes are exported:
+
+* `id` - The ID of the Healthcare Med Tech Service.
+
+* `identity` - The `identity` block as defined below.
+
+* `eventhub_namespace_name` - The namespace name of the Event Hub of the Healthcare Med Tech Service.
+
+* `eventhub_name` - The name of the Event Hub of the Healthcare Med Tech Service.
+
+* `eventhub_consumer_group_name` - The Consumer Group of the Event Hub of the Healthcare Med Tech Service.
```

Comment:
could these all not be combined into eventhub_consumergroup_id?

Reply:
I'm afraid that we can't since the eventhub_consumergroup_id cannot be put back just by the properties we get back from the service. I confirmed this behavior with the upstream service team and there is no plan of changing.

Reply:
we can't since the consumer_id cannot be put back together by using the information returned by the service.

we are not able to get the resource group name of the consumer group.

Reply:
The user can use eventhubs from another resource groups and the service won't returned the rg name back to us, so we are not able to put the id back by lacking this important part.

Reply:
i'm sorry but i don't folloiw, the consumer group will always be within the same eventhub namespace/name specified? these three properties all need to be for the same evenhubt consumer group, ?

Reply:
No worries, the reason why I'm not using the id is we are not able to get the id back from the service team and this blocks us from setting the eventhub information back to state. 

user can set the consumer group from another resource group, but we don't know which resource group is when retrieving data from the api in read function. 

If it's still not clear to you, below is the model in SDK:
```
type IotEventHubIngestionEndpointConfiguration struct {
	// EventHubName - Event Hub name to connect to.
	EventHubName *string `json:"eventHubName,omitempty"`
	// ConsumerGroup - Consumer group of the event hub to connected to.
	ConsumerGroup *string `json:"consumerGroup,omitempty"`
	// FullyQualifiedEventHubNamespace - Fully qualified namespace of the Event Hub to connect to.
	FullyQualifiedEventHubNamespace *string `json:"fullyQualifiedEventHubNamespace,omitempty"`
}
```
as you can see, we can't get the id back and thus user needs to explicitly specify all the eventhub related information.

Reply:
what resource group will that create an eventhub in? the same on as the med tech service?

Reply:
this is the point, the medTech service can refer to an eventhub that exists in a different RG and we won't be able to get this RG from the api.

Reply:
how does it know what resource group it is in when all we specify is the name?

Reply:
it doesn't need to know the RG name, if you check the portal, you can see the service list all the available eventhub namespaces in the dropdown list:
![image](https://user-images.githubusercontent.com/92154856/183684532-0a608038-48cf-48f9-8d43-7aa7663bb145.png)

I aslo checked with them about this before, they said they don't validate the eventhub from the backend at the current stage...

Reply:
so what they are doing is grabbing ALL the evenhubs in the subscription. what happens if there are two event hubs with the exact same name across two different RGs? how does it know what to use? that seems like a bug in the API

Reply:
From the azure portal, the EventHub will be listed based on the namespace that selected by the user. However, there isn't a validation of EventHub when calling from the API. I confirmed this behavior with the service team and they said the current design is to validate the EventHub is well-formed only, as for the validation of whether the endpoint is active or not, it happens at the runtime. The designe is to ensure that service functions correctly if the customer modifies or deletes the Event Hub after our service is deployed.

Reply:
how does a terraform user with 2 evenhubs with the same name across 2 different resource groups specify which one to use? what happens then?

Reply:
After some internal testing and discussion we determined that namespaces are globally unique so we shouldn't run into this issue